### PR TITLE
TTL Functional Tests

### PR DESCRIPTION
**Pull request by mrzenioszeniou**
Introduced some TTL tests to complement #3504.

* Added a simple unit test in to ensure transactions are removed as soon as they're expired.
* Added JCLI test to ensure no transactions can be created without a time validity.
* Added JCLI test to ensure different time validity yields different transaction.
* Added single-node test to ensure expired transactions are rejected.
* Added single-node test to ensure transactions with long TTLs are rejected.

### CODE REVIEW

File: testing/jormungandr-integration-tests/src/jormungandr/transactions.rs
Comment (Zeegomo) [inline testing/jormungandr-integration-tests/src/jormungandr/transactions.rs:124]: mmm, I think there's still some date magic under this and I don't particularly like it. Didn't we fix it already (or was it something different)? IIRC we agreed to not have date manipulation, at least for interfaces which are simply building a transaction like you would on the command line like `Wallet`

File: testing/jormungandr-integration-tests/src/jormungandr/transactions.rs
Comment (Zeegomo) [inline testing/jormungandr-integration-tests/src/jormungandr/transactions.rs:203]: Same as above, this is also confusing, why would it fail if the validity date is set to the max allowed?

File: testing/jormungandr-testing-utils/src/testing/fragments/sender.rs
Comment (Zeegomo) [inline testing/jormungandr-testing-utils/src/testing/fragments/sender.rs:145]: This sounds like validity is an optional field, and indeed we have also `send_transaction`. I would pick another less ambiguous name, and also understand why we have these two methods. If `send_transaction` is a convenience method to do some date magic it could just pick the value and reuse `send_transaction_with_validity`

File: testing/jormungandr-testing-utils/src/testing/fragments/sender.rs
Comment (Zeegomo) [inline testing/jormungandr-testing-utils/src/testing/fragments/sender.rs:155]: This is misleading, the wallet treats the date as "current date" (poor choice of name there, besides the unwanted manipulation), so this does not set the expiry date for the transaction to `valid_until`, but to the next epoch.
I think this is another evidence that `Wallet` should not do anything with the date it receives, and use it directly to set the `valid_until` transaction field

**Review by Zeegomo — CHANGES_REQUESTED** [review]
The tests themselves are good, but in my opinion the way the test framework is written introduces unnecessary confusion.
I would rather fix that first than keep adding stuff on top of it which we would need to be fixed later

File: testing/jormungandr-integration-tests/src/jormungandr/transactions.rs
Comment (mrzenioszeniou) [reply testing/jormungandr-integration-tests/src/jormungandr/transactions.rs:124]: Yes that is correct. In this case the magic is hidden in `jormungandr_testing_utils::testing::fragments::FragmentBuilder`. We have #3529 which should clean these up, so I didn't bother fixing them again. We can rebase this branch onto it or the other way around.

File: testing/jormungandr-testing-utils/src/testing/fragments/sender.rs
Comment (mrzenioszeniou) [reply testing/jormungandr-testing-utils/src/testing/fragments/sender.rs]: Oh my bad, that's leftover code from an attempt to clean up the testing API. I got rid of it. `BlockDate` magic aside, the validity for a transaction is defined upon construction of the `FragmentSender` and all transactions created by it use that `BlockDate`.

File: testing/jormungandr-integration-tests/src/common/jcli/services/fragment_check.rs
Comment (mrzenioszeniou) [reply testing/jormungandr-integration-tests/src/common/jcli/services/fragment_check.rs]: Often enough the tests end with an assertion, which is why some these assertive functions consume `self`. You're right though, there's no reason why they should. I changed them to `&self`.

File: testing/jormungandr-integration-tests/src/jormungandr/transactions.rs
Comment (Zeegomo) [reply testing/jormungandr-integration-tests/src/jormungandr/transactions.rs:124]: Indeed, I recalled seeing a PR for that. All good then! But I would wait for it to land before merging this PR, as it will require changes to the tests I guess

File: jormungandr/src/fragment/pool.rs
Comment (eugene-babichenko) [inline jormungandr/src/fragment/pool.rs:678]: Unrelated: this is funny, that such no-op transaction can be built.

**Review by eugene-babichenko — APPROVED** [review]
All looking good. I merged #3529, so please rebase onto the latest `master`.

**Review by Zeegomo — APPROVED** [review]

**Pull request merged**
